src/test/java/bio/terra/app/controller/DatasetsApiControllerTest.java

```diff
@@ -239,6 +245,61 @@ void testUpdateSnapshotBuilderSettings() throws Exception {
     verify(snapshotBuilderService).updateSnapshotBuilderSettings(DATASET_ID, SETTINGS);
   }
 
+  @Test
+  void testCreateCriteriaData() throws Exception {
+    TestUtils.mapFromJson(
+        """
+        {"kind":"list","name":"name","id":0}""", SnapshotBuilderCriteria.class);
+
+    TestUtils.mapFromJson(
+        """
+        {"kind":"list","name":"name","id":0,"values":[]}""",
+        SnapshotBuilderProgramDataListCriteria.class);
+
+    TestUtils.mapFromJson(
+        """
+        {"kind":"range","name":"name","id":0,"low":0,"high":10}""",
+        SnapshotBuilderProgramDataRangeCriteria.class);
+  }
+
+  @Test
+  void testCreateSnapshotRequest() throws Exception {
+    mockValidators();
+    SnapshotBuilderAccessRequest expected =
+        new SnapshotBuilderAccessRequest()
+            .name("name")
+            .researchPurposeStatement("purpose")
+            .datasetRequest(
+                new SnapshotBuilderRequest()
+                    .addCohortsItem(
+                        new SnapshotBuilderCohort()
+                            .name("cohort")
+                            .addCriteriaGroupsItem(new SnapshotBuilderCriteriaGroup()))
```

Comment:
Can you add at least one criteria to the group? It'd be good to have another check for round-trip operations on the `oneOf` objects.